### ISSUE TITLE
Template string test titles are considered invalid with v4.12.0

### ISSUE THREAD

**Issue by benderTheCrime** [lo1tuma/eslint-plugin-mocha#157]
Thanks for the useful plugin!

Since upgrading to v4.12.0, I've been getting eslint errors associated with some test titles that are template literals.

Example (latest eslint, eslint-plugin-mocha):

.eslintrc.json:
```json
{
  "plugins": [
    "mocha"
  ],
  "rules": {
    "mocha/valid-test-description": 2
  }
}
```

This passes:
```javascript
it('should be a vaild test title', function() {});
```

This fails:
```javascript
it(`should be a vaild test title`, function() {}); // notice the ticks
```

I think the issue has something to do with this line:
https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/lib/rules/valid-test-description.js#L23

I'm not sure if there is an AST util for string template literals, or not, but the `StringLiteral` method name seems suspect.

**Comment by lo1tuma** [comment]
Thanks for reporting this issue @benderTheCrime. You are right, this happens due the change in the line you mentioned. I‘ve refactored a lot of code and it relied on the test-suite to avoid breaking changes, but it turns out there is no test-case for `valid-test-description` with template strings. I will add one and fix this problem asap.

**Comment by benderTheCrime** [comment]
Awesome, thanks!

**Comment by lo1tuma** [comment]
I’ve just took a closer look at how it was implemented before. Template strings or any other non string literals that are passed to first argument of e.g. `it()` were just skipped and not reported at all (see [here](https://github.com/lo1tuma/eslint-plugin-mocha/blob/91d608b43a9e1c48a22a3e801135ae93919dc283/lib/rules/valid-test-description.js#L21). I’m going to restore this behavior for now. Maybe it would make sense to create a follow-up issue to discuss whether or not it makes sense to (1) disallow certain types of arguments in `it()` and (2) if we could somehow validate template strings, because they could contain dynamic parts.

**Comment by lo1tuma** [comment]
I’ve published `4.12.1` which should fix this problem. Please let me know if you still have problems with this version.
